api/src/modules/materials/material.repository.ts

```diff
@@ -1,10 +1,62 @@
-import { EntityRepository } from 'typeorm';
+import { EntityRepository, SelectQueryBuilder } from 'typeorm';
 import { Material } from 'modules/materials/material.entity';
 import { ExtendedTreeRepository } from 'utils/tree.repository';
 import { CreateMaterialDto } from 'modules/materials/dto/create.material.dto';
+import { Logger, NotFoundException } from '@nestjs/common';
+import { SourcingLocation } from 'modules/sourcing-locations/sourcing-location.entity';
 
 @EntityRepository(Material)
 export class MaterialRepository extends ExtendedTreeRepository<
   Material,
   CreateMaterialDto
-> {}
+> {
+  logger: Logger = new Logger(MaterialRepository.name);
+
+  /**
+   * @description Retrieves materials and it's ancestors (in a plain format) there are registered sourcingLocations for
+   */
+
+  async getSourcingDataMaterialsWithAncestry(): Promise<Material[]> {
+    // Join and filters over materials present in sourcing-locations. Resultant query returns IDs of elements meeting the filters
+    const queryBuilder: SelectQueryBuilder<Material> = this.createQueryBuilder(
+      'm',
+    )
+      .select('m.id')
+      .innerJoin(SourcingLocation, 'sl', 'sl.materialId = m.id')
+      .distinct(true);
+
```

Comment:
In case it's not clear, I am not retrieving the query and params straight off the chain of createQueryBuilder because we will need to add additional filters here in the future, so we easily do this like:

if(condition)
  queryBuilder.andWhere(...)